neuraxle/metaopt/tpe.py

```diff
@@ -51,7 +52,8 @@ def find_next_best_hyperparams(self, auto_ml_container: AutoMLContainer) -> Hype
         :rtype: HyperparameterSamples
         """
         # Flatten hyperparameter space
-        flat_hyperparameter_space: HyperparameterSpace = auto_ml_container.hyperparameter_space.to_flat()
+        hyperparams_space_list: List[(str,HyperparameterDistribution)] = list(auto_ml_container.hyperparameter_space.to_flat_dict().items())
```

Comment:
Does this breaks encapsulation ? Getters should be used to avoid using the inner hyperparams and inner space directly. 
```suggestion
        hyperparams_space_list: List[(str,HyperparameterDistribution)] = list(auto_ml_container.get_hyperparameter_space().to_flat_dict().items())
```

Reply:
The hyperparameter_space attribute of this AutoMlContainer instance is already the result from a get_hyperparameter_space(). I don't think calling get_hyperparameter_space() works on an AutoMlContainer instance anyway, since it does not inherit from BaseTransformer.